torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -208,7 +208,7 @@ c10::optional<MatchedSchema> tryMatchSchema(
       self = c10::nullopt;
     } else if (!arg.kwarg_only() && used_args < args.size()) {
       // allow zeros(IntList sizes) to work with zeros(1, 2) or zeros(1)
-      if (arg.type()->kind() == TypeKind::ListType && // the formal must be a list
+      if (allow_conversions && arg.type()->kind() == TypeKind::ListType && // the formal must be a list
```

Comment:
what's this for?

Reply:
This is a type conversion from (int a, int b, int c) -> []int. Just as the other type conversions, it should only be run on the second pass when allow_conversions is true. I think you have an issue raised about this logic IIRC.

I ran into a bug bc of it when testing this.

Reply:
nice! Can you add a test that checks this to avoid regression in the future?

Reply:
Hm i think the error actually was in a different PR, the torch.tensor one and accidentally made it's way in here. Removing from this PR and adding a test in that one